### PR TITLE
Added more tests for tools

### PR DESCRIPTION
**Pull request by jelonanX**
Added new symlink and path validate tests for tools fpgaconf and coreidle.

### CODE REVIEW

File: testing/fpgaconf/test_fpgaconf_c.cpp [hunk 1]
```diff
@@ -707,16 +843,34 @@ TEST_P(fpgaconf_c_p, embed_nullchar1) {
...
   strcpy(three, "-B");
   strcpy(four, "0x5e");
-  strcpy(five, "copy_bitstream\0.gbs");
 
   char *argv[] = { zero, one, two, three, four,
                    five };
-  EXPECT_NE(fpgaconf_main(6, argv), 0);
-  unlink(copy_gbs_.c_str());
+
+  if (!symlink(copy_gbs_.c_str(), symlink_gbs.c_str())) {
```
Comment (tswhison) [inline testing/fpgaconf/test_fpgaconf_c.cpp]: Do we want to fail the test if this fails?  Not sure..

File: testing/fpgaconf/test_fpgaconf_c.cpp [hunk 2]
```diff
@@ -728,12 +882,32 @@ TEST_P(fpgaconf_c_p, embed_nullchar2) {
...
   strcpy(four, "0x5e");
-  strcpy(five, "\0copy_bitstream.gbs");
 
   char *argv[] = { zero, one, two, three, four,
                    five };
-  EXPECT_NE(fpgaconf_main(6, argv), 0);
-  unlink(copy_gbs_.c_str());
+
+  // Create link directories
+  if (!(mkdir("./link1", S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH))
```
Comment (tswhison) [inline testing/fpgaconf/test_fpgaconf_c.cpp]: Fail if the mkdir or symlink calls fail?

File: testing/fpgaconf/test_fpgaconf_c.cpp [hunk 3]
```diff
@@ -750,8 +769,8 @@ TEST_P(fpgaconf_c_p, encoding_path) {
  * @test       relative_path
  * @brief      Test: fpgaconf_main
  * @details    When gbs file locates in parent directory and command params<br>
- *             contains path traversal, fpgaconf_main displays file<br>
- *             error and returns non-zero.<br>
+ *             contains path traversal. On success, fpgaconf_main loads bitstream<br>
+ *             and returns zero. Otherwise, it displays file error and returns non-zero.<br>
  */
 TEST_P(fpgaconf_c_p, relative_path) {
```
Comment (tswhison) [inline testing/fpgaconf/test_fpgaconf_c.cpp]: I thought we would expect this one to fail?

File: testing/fpgaconf/test_fpgaconf_c.cpp (hunk 2 above)
Comment (jelonanX) [reply testing/fpgaconf/test_fpgaconf_c.cpp]: I believe so. But I have yet figure out how to get mkdir and symlink to fail. Any suggestions?

File: testing/fpgaconf/test_fpgaconf_c.cpp (hunk 3 above)
Comment (jelonanX) [reply testing/fpgaconf/test_fpgaconf_c.cpp]: Previously, it was a false positive on set_userclk. Setup has been modified to load in a valid bitstream so this test succeeds now.

Comment (tswhison) [reply testing/fpgaconf/test_fpgaconf_c.cpp]: Ah, cool.

**Pull request merged**